src/ui/pages/AssetInfo/AssetHeader.tsx

```diff
@@ -1,42 +1,192 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 import { useCurrency } from 'src/modules/currency/useCurrency';
 import type { Asset } from 'src/modules/zerion-api/requests/asset-get-fungible-full-info';
 import { formatPriceValue } from 'src/shared/units/formatPriceValue';
-import { emDash } from 'src/ui/shared/typography';
+import VerifiedIcon from 'jsx:src/ui/assets/verified.svg';
 import { HStack } from 'src/ui/ui-kit/HStack';
 import { TokenIcon } from 'src/ui/ui-kit/TokenIcon';
 import { UIText } from 'src/ui/ui-kit/UIText';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import { emitter } from 'src/ui/shared/events';
+import { emDash } from 'src/ui/shared/typography';
+import { useWindowSizeStore } from 'src/ui/shared/useWindowSizeStore';
+import * as styles from './styles.module.css';
 
-export function AssetHeader({
-  asset,
-  className,
-}: {
-  asset: Asset;
-  className?: string;
-}) {
+export function AssetHeader({ asset }: { asset: Asset }) {
   const { currency } = useCurrency();
+  const { innerWidth } = useWindowSizeStore();
+  const isUntrackedAsset = asset.meta.price == null;
+  const priceElementRef = useRef<HTMLDivElement>(null);
+  const symbolPlaceholderElementRef = useRef<HTMLDivElement>(null);
+  const dashPlaceholderElementRef = useRef<HTMLDivElement>(null);
+  const pricePlaceholderElementRef = useRef<HTMLDivElement>(null);
+
+  const [iconOffset, setIconOffset] = useState(0);
+  const [symbolOffset, setSymbolOffset] = useState(0);
+  const [dashOffset, setDashOffset] = useState(0);
+  const [priceOffset, setPriceOffset] = useState(0);
+
+  useEffect(() => {
+    return emitter.on('assetPriceSelected', (formattedPrice) => {
+      if (priceElementRef.current) {
+        priceElementRef.current.innerText = formattedPrice;
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (
+      symbolPlaceholderElementRef.current &&
+      pricePlaceholderElementRef.current &&
+      dashPlaceholderElementRef.current
+    ) {
+      const iconWidth = 20;
+      const gapWidth = 4;
+      const panelWidth = innerWidth - 92;
+      const symbolWidth = symbolPlaceholderElementRef.current.offsetWidth;
+      const dashWidth = dashPlaceholderElementRef.current.offsetWidth;
+      const priceWidth = pricePlaceholderElementRef.current.offsetWidth;
+      const contentWidth =
+        iconWidth + symbolWidth + dashWidth + priceWidth + 4 * gapWidth;
+
+      const startOffset = (panelWidth - contentWidth) / 2;
+      setIconOffset((startOffset - iconWidth) * 2);
```

Comment:
These offset-states always change atomically together and never independently
```
    setOffsets({
      iconOffset: (startOffset - iconWidth) * 2,
      symbolOffset: startOffset + iconWidth + 2 * gapWidth - 56,
      dashOffset: startOffset + iconWidth + 2 * gapWidth + symbolWidth + gapWidth,
      priceOffset:
        startOffset +
        iconWidth +
        2 * gapWidth +
        symbolWidth +
        gapWidth +
        dashWidth +
        gapWidth,
    });
```
forget about my suggestion if any of these offsets might need to change independently in the future :)